From 8efbc3b5e8ff7029fb451e45448c9e54d052253b Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ilies.chergui@gmail.com>
Date: Sat, 8 Apr 2023 15:53:32 +0100
Subject: [PATCH 4/5] lib/tools-common.in: align cryptsetup args with NVidia

The L4T disk_encryption_helper.func uses aes-xts-plain64:sha256 cipher and
256 bits key-size.

Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>
---
 lib/tools-common.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/tools-common.in b/lib/tools-common.in
index 749b1fa..0efef64 100644
--- a/lib/tools-common.in
+++ b/lib/tools-common.in
@@ -459,11 +459,12 @@ encrypt_one_partition() {
     local name="$3"
     local realname="$4"
     echo "Encrypting $luksdev($name)..."
-    if ! cat "$ppdir/passphrase" | cryptsetup --type luks --cipher aes-xts-plain64 --hash sha256 \
+    if ! cat "$ppdir/passphrase" | cryptsetup --type=luks2 --cipher=aes-xts-plain64 --hash=sha256 --key-size=256 \
 					      --use-random --key-file - luksFormat $luksdev >/dev/null 2>&1; then
 	echo "[FAIL $luksdev($name)]"
 	return 1
     fi
+
     # now 'open' it
     if ! cryptsetup luksOpen --key-file "$ppdir/passphrase" $luksdev $realname 2>/dev/null; then
 	echo "[$luksdev($name) FAIL]"
@@ -738,7 +739,7 @@ initialize_devices() {
     local fakept=`mktemp -q`
     if make_fake_parttable $fakept; then
 	local ppdir=`mktemp -q -d /run/setup-pp.XXXXXX`
-	/usr/sbin/nvluks-srv-app --get-unique-pass --context-string=@DISK_ENCRYPTION_CONTEXT@ > $ppdir/passphrase
+	/usr/sbin/nvluks-srv-app --get-unique-pass --context-string=@DISK_ENCRYPTION_CONTEXT@ > $ppdir/passphrase && /bin/chmod 0600 $ppdir/passphrase
 	if ! create_luks_partitions $ppdir $fakept; then
 	    rm -rf $ppdir
 	    return 1
-- 
2.25.1

